src/providers/eth_provider/starknet/relayer.rs

```diff
@@ -28,24 +27,24 @@ static RELAYER_SIGNER: LazyLock<LocalWallet> = LazyLock::new(|| {
 /// A relayer holding a lock on a mutex on an account and connected to the Starknet network.
 /// The relayer is used to sign  transactions and broadcast them on the network.
 #[derive(Debug)]
-pub struct LockedRelayer<'a> {
+pub struct LockedRelayer<'a, SP: Provider + Send + Sync> {
```

Comment:
do we need SP here? I don't think we will ever use something else than `JsonRpcClient<HttpTransport>`

Reply:
Mhhh I think so, maybe once this is stable we can have a look at how to simplify but in the `deploy_evm_contract`, I had to use:

```rust
// Relay the transaction
        let starknet_transaction_hash = LockedRelayer::new(
            current_nonce.lock().await,
            relayer.address(),
            relayer_balance,
            self.starknet_provider(),
            self.starknet_provider().chain_id().await.expect("Failed to get chain id"),
        )
        .relay_transaction(&tx_signed)
        .await
        .expect("Failed to relay transaction");
```

which involved the starknet provider which is SP, that's why I had to put a generic here...